src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticListener.cs

```diff
@@ -353,4 +359,45 @@ public void Dispose()
 
         #endregion
     }
+
+    /// <summary>
+    /// Current this EventSource is only here so that the Debugger can inject code using Function evaluation
+    /// We may add actual logging requests as well at some point.   This can be removed if the debugger  no 
+    /// longer needs it.   
+    /// </summary>
+    [EventSource(Name = "Microsoft-Diagnostics-DiagnosticSource")]
+    class DiagnosticSourceEventSource : EventSource
+    {
+        static public DiagnosticSourceEventSource Logger = new DiagnosticSourceEventSource();
+
+        /// <summary>
+        /// On every command (which the debugger can force by turning on this EventSource with ETW)
+        /// call a function that the debugger can hook to do an arbitrary func evaluation.  
+        /// </summary>
+        /// <param name="args"></param>
+        protected override void OnEventCommand(EventCommandEventArgs args)
+        {
+            BreakPointWithDebuggerFuncEval();
```

Comment:
Yes, command only occur when something outside (e.g. a Listener or ETW), sends one.   They are very rare.   Frankly we only expect them on debugger attach.